hack/consecutive-upgrade-test-operator-sdk.sh

```diff
@@ -0,0 +1,323 @@
+  else
+    ./hack/retry.sh 10 30 "[[ \$(${CMD} get vmi -l kubevirt.io/outdatedLauncherImage -A --no-headers | wc -l) -eq 0 ]]" "${CMD} get vmi -l kubevirt.io/outdatedLauncherImage -A"
+    echo "All the running VMs got upgraded"
+  fi
+
+  ./hack/retry.sh 5 30 "~/virtctl stop testvm -n ${VMS_NAMESPACE}"
+  ${CMD} delete vm -n ${VMS_NAMESPACE} testvm
+
+  KUBECTL_BINARY=${CMD} ./hack/test_quick_start.sh
+
+  Msg "Check that OVS is deployed or not deployed according to deployOVS annotation in HCO CR."
+  ./hack/retry.sh 40 15 "CMD=${CMD} PREVIOUS_OVS_ANNOTATION=${PREVIOUS_OVS_ANNOTATION}\
+   PREVIOUS_OVS_STATE=${PREVIOUS_OVS_STATE} ./hack/check_upgrade_ovs.sh"
+
+  Msg "Ensure that console plugin deployment and service has been renamed successfully"
+  KUBECTL_BINARY=${CMD} INSTALLED_NAMESPACE=${HCO_NAMESPACE} ./hack/check_upgrade_console_plugin.sh
+
+  Msg "Check that the v2v CRDs and deployments were removed"
+  if ${CMD} get crd | grep -q v2v.kubevirt.io; then
+      echo "The v2v CRDs should not be found; they had to be removed."
+      exit 1
+  else
+      echo "v2v CRDs removed"
+  fi
+  if ${CMD} get deployments -n ${HCO_NAMESPACE} | grep -q vm-import; then
+      echo "v2v deployments should not be found; they had to be removed."
+      exit 1
+  else
+      echo "v2v deployments removed"
+  fi
+
+  Msg "Check that the v2v references were removed from .status.relatedObjects"
+  if ${CMD} -n ${HCO_NAMESPACE} ${HCO_KIND} ${HCO_RESOURCE_NAME} -o=jsonpath={.status.relatedObjects[*].apiVersion} | grep -q v2v.kubevirt.io; then
+      echo "v2v references should not be found in relatedObjects; they had to be removed."
+      exit 1
+  else
+      echo "v2v references removed from .status.relatedObjects"
+  fi
+
+  Msg "Check that the TTO CRD was removed"
+  if ${CMD} get crd | grep -q tektontasks.tektontasks.kubevirt.io; then
+      echo "The TTO CRD should not be found; it had to be removed."
+      exit 1
+  else
+      echo "TTO CRD removed"
+  fi
+
+  Msg "Check that the TTO references were removed from .status.relatedObjects"
+  if ${CMD} -n ${HCO_NAMESPACE} ${HCO_KIND} ${HCO_RESOURCE_NAME} -o=jsonpath={.status.relatedObjects[*].apiVersion} | grep -q tektontasks.kubevirt.io; then
+      echo "TTO reference should not be found in relatedObjects; it has to be removed."
+      exit 1
+  else
+      echo "TTO reference removed from .status.relatedObjects"
+  fi
+
+  Msg "check virtio-win image is in configmap"
+  VIRTIOWIN_IMAGE_CSV=$(${CMD} get ${CSV} -n ${HCO_NAMESPACE} \
+    -o jsonpath='{.spec.install.spec.deployments[?(@.name=="hco-operator")].spec.template.spec.containers[0].env[?(@.name=="VIRTIOWIN_CONTAINER")].value}')
+  VIRTIOWIN_IMAGE_CM=$(${CMD} get cm virtio-win -n ${HCO_NAMESPACE} -o jsonpath='{.data.virtio-win-image}')
+
+  [[ "${VIRTIOWIN_IMAGE_CSV}" == "${VIRTIOWIN_IMAGE_CM}" ]]
+
+}
+
+upgrade $INITIAL_VERSION $MID_VERSION $OO_MID_BUNDLE
+upgrade $MID_VERSION $TARGET_VERSION $OO_LAST_BUNDLE
```

Comment:
I'm not sure about this implementation. 

let's say we're doing upgrade from x-->y-->z

It's over kill to run all the test and checks after both x-->y and y-->z, first because these upgrades are tested elsewhere, but also because what will be more interesting, for example, is to start a VM after the x, and validate it after z. In addition, some of the checks are specific for version x to version y upgrade, e.g. the removal of TTO.

I think that the script should:
1. check that the installation of x is ok
2. start a vm
3. upgrade to y
4. check that the upgrade was done. Check that the VM is still running. check the operator condition
5. upgrade to z
6. check all the above (in 4) + run the func tests and all the relevant checks.

Reply:
Makes sense. I put in all that within the function because I wasn't sure how much is enough. Does b06a052 seem more appropriate in terms of scope?